app/models/user_group.rb

```diff
@@ -60,7 +51,7 @@ def users
   end
 
   def active_users
-    self.active_roles.map { |role| role.user }
+    self.active_roles.includes(:user).map(&:user)
```

Comment:
I think you can realise this as a relation above.

```ruby
has_many :users, through: :roles
has_many :active_users, through: :active_roles
```

Reply:
Thanks. `has_many :active_users, through: :active_roles, source: :user, class_name: "User"` worked.